Xamarin.Forms.Platform.Android/Renderers/ShellToolbarTracker.cs

```diff
@@ -286,6 +286,20 @@ protected virtual async Task UpdateDrawerArrow(Context context, Toolbar toolbar,
 				_drawerToggle.DrawerIndicatorEnabled = false;
 			}
 			_drawerToggle.SyncState();
+
+			//this needs to be set after SyncState
+			UpdateToolbarIconAccessibilityText(toolbar, _shellContext.Shell);
+		}
+
+		protected virtual void UpdateToolbarIconAccessibilityText(Toolbar toolbar, Shell shell)
+		{
+			var shellIconTextDescription = shell.FlyoutIcon?.AutomationId ?? AutomationProperties.GetHelpText(_shellContext.Shell.FlyoutIcon) ?? shell.AutomationId;
+
+			//setting a empty text will cause to not read the default ´OK´
+			if (string.IsNullOrEmpty(shellIconTextDescription))
+				return;
+
+			toolbar.NavigationContentDescription = shellIconTextDescription;
```

Comment:
Should this be
```
toolbar.SetNavigationContentDescription(_shellContext.Shell.FlyoutIcon, _defaultNavigationContentDescription);
```

Reply:
I kept the setting AutomationID so users could use this on UITests